include/crocoddyl/core/activations/quadratic-flat.hpp

```diff
@@ -0,0 +1,145 @@
+// BSD 3-Clause License
+//
+// Copyright (C) 2020, LAAS-CNRS, University of Edinburgh
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_ACTIVATIONS_QUADRATIC_FLAT_HPP_
+#define CROCODDYL_CORE_ACTIVATIONS_QUADRATIC_FLAT_HPP_
+
+#include <iostream>
+#include <stdexcept>
+#include "crocoddyl/core/fwd.hpp"
+#include "crocoddyl/core/activation-base.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+
+namespace crocoddyl {
+
+/**
+ * @brief Quadratic-flat activation
+ *
+ * This activation function describes a quadratic exponential activation depending on the square norm
+ * of a residual vector, i.e. \f[ \begin{equation} 1 - exp(\|\mathbf{r}\|^2 / \alpha) \end{equation}
+ * \f] where \f$\alpha\f$ defines the width of the quadratic basin, \f$r\f$ is the scalar residual,
+ * \f$nr\f$ is the dimension of the residual vector. Far
+ * away from zero, the quadFlat activation is nearly flat.
+ *
+ * The computation of the function and it derivatives are carried out in
+ * `calc()` and `caldDiff()`, respectively.
+ *
+ * \sa `calc()`, `calcDiff()`, `createData()`
+ */
+template <typename _Scalar>
+class ActivationModelQuadFlatTpl : public ActivationModelAbstractTpl<_Scalar> {
+ public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  typedef _Scalar Scalar;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef ActivationModelAbstractTpl<Scalar> Base;
+  typedef ActivationDataAbstractTpl<Scalar> ActivationDataAbstract;
+  typedef ActivationDataQuadFlatTpl<Scalar> Data;
+  typedef typename MathBase::VectorXs VectorXs;
+  typedef typename MathBase::MatrixXs MatrixXs;
+
+  /**
+   * @brief Initialize the quadratic-flat activation model
+   *
+   * The default `alpha` value is defined as 1.
+   *
+   * @param[in] nr     Dimension of the residual vector
+   * @param[in] alpha  Width of quadratic basin (default: 1.)
+   */
+  explicit ActivationModelQuadFlatTpl(const std::size_t &nr, const Scalar &alpha = Scalar(1.))
+      : Base(nr), alpha_(alpha) {
+    if (alpha < Scalar(0.)) {
+      throw_pretty("Invalid argument: "
+                   << "alpha should be a positive value");
+    }
+  };
+  virtual ~ActivationModelQuadFlatTpl(){};
+
+  /**
+   * @brief Compute the quadratic-flat function
+   *
+   * @param[in] data  Quadratic-flat activation data
+   * @param[in] r     Residual vector \f$\mathbf{r}\in\mathbb{R}^{nr}\f$
+   */
+  virtual void calc(const boost::shared_ptr<ActivationDataAbstract> &data, const Eigen::Ref<const VectorXs> &r) {
+    if (static_cast<std::size_t>(r.size()) != nr_) {
+      throw_pretty("Invalid argument: "
+                   << "r has wrong dimension (it should be " + std::to_string(nr_) + ")");
+    }
+    boost::shared_ptr<Data> d = boost::static_pointer_cast<Data>(data);
+
+    d->a0 = exp(-r.squaredNorm() / alpha_);
+    data->a_value = Scalar(1.0) - d->a0;
```

Comment:
The implementation is done as 1-exp(-||r||^2), while documentation mentions 1-exp(||r||^2)